backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -1,48 +1,45 @@
 package com.google.sps.data;
 
+import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
 import java.time.ZonedDateTime;
 
-/** A user who want to be matched. */
+/** A user who wants to be matched. */
 public final class Participant {
 
-  /** Datastore ID */
-  private final long id;
   /** Google username (ldap) */
   private final String username;
-  /** Time user is starts being available */
-  private final ZonedDateTime startTimeAvailable;
+  /** Time user starts being available */
+  // TODO: make these fields final once Participant and User classes are created
+  private ZonedDateTime startTimeAvailable;
   /** Time user is available until */
-  private final ZonedDateTime endTimeAvailable;
+  private ZonedDateTime endTimeAvailable;
   /** How long user wants to chat */
-  private final int duration;
+  private int duration;
+  /** Current match in datastore, null if never found a match or already been returned */
+  private long currentMatchId;
   /** Time of submitted form */
   private final long timestamp;
 
   /** Initialize constructor fields */
   public Participant(
-      long id,
       String username,
       ZonedDateTime startTimeAvailable,
       ZonedDateTime endTimeAvailable,
       int duration,
+      long currentMatchId,
```

Comment:
Is it possible that when we create a Participant, we already know the match id? If not, should this field be always initialized to 0?

Reply:
When we create a participant just to pass to the datastore to add a participant, yes, matchId would be 0. However, when we return a Participant (from a username key that leads you to an entity whose properties are the Participant's fields), we definitely want to return the matchId in the datastore, so no need to always initialize to 0 if you'd also have to setMatchId() separately too.

Reply:
Got it. I don't know if Java supports default value for a field in a constructor. Or we should create a new constructor, or just update javadoc to let the user know that it's OK to assign 0 to this field if matchId is unknown during construction.

Reply:
I'll add a javadoc!